sklearnex/tests/test_run_to_run_stability.py

```diff
@@ -154,6 +155,8 @@ def test_standard_estimator_stability(estimator, method, dataframe, queue):
         pytest.skip(f"variation observed in {estimator}.score")
     if estimator in ["IncrementalEmpiricalCovariance"] and method == "mahalanobis":
         pytest.skip("allowed fallback to sklearn occurs")
+    if not _IS_INTEL and ("Neighbors" in estimator or "LocalOutlierFactor" in estimator) and method in ["score", "predict", "kneighbors", "kneighbors_graph"]:
+        pytest.skip(f"{estimator} shows instability in Non-Intel hardware")

```

Comment:
I propose to make a common lambda for checking this condition as the nature of these crashes is the same